src/BatchInterceptor.ts

```diff
@@ -31,13 +31,22 @@ export class BatchInterceptor<
     BatchInterceptor.symbol = Symbol(options.name)
     super(BatchInterceptor.symbol)
     this.interceptors = options.interceptors
+
+    /**
+     * Ensure that child interceptors have no event listeners
+     * when the batch interceptor is constructed. The listeners may
+     * persists between Fast Refresh in modern apps, causing
+     * the same intercepted request to be handled multiple times.
+     * @see https://github.com/mswjs/msw/issues/1271
+     */
+    for (const interceptor of this.interceptors) {
+      interceptor['emitter'].pruneListeners()
```

Comment:
Coupling internal `_events_` pruning and `.removeAllListeners()` causes issues. Looks like we cannot remove the internal record all the time when the consumer removes listeners.

I'm adding a designated method that'd do both only in this particular scenario.